File: client-web/src/types/queue.ts
export interface QueueAnalysisRequest {
  lambda_rate: number
  mu_rate: number
}

export interface QueueAnalysisResponse {
  lambda_rate: number
  mu_rate: number

  arrival_rate_unit: string
  service_rate_unit: string
  time_unit: string

  is_stable: boolean

  utilization: number
  utilization_percent: number

  average_orders_in_system: number | null
  average_waiting_time: number | null
  average_waiting_time_hours: number | null
  average_time_in_system: number | null
  average_time_in_system_hours: number | null

  conclusion: string
}

export interface QueueFormulasResponse {
  model_name: string
  description: string
  stability_condition: string
  formulas: Record<string, string>
}

File: server/app/api/v1/queue.py
from fastapi import APIRouter

from app.schemas.queue import (
    QueueAnalysisRequest,
    QueueAnalysisResponse,
    QueueFormulasResponse,
)
from app.services.queue_analysis import analyze_queue


router = APIRouter(prefix="/queue", tags=["queue-analysis"])


@router.post(
    "/analyze",
    response_model=QueueAnalysisResponse,
    summary="Выполнить анализ системы массового обслуживания",
    description=(
        "Принимает интенсивность поступления заказов λ и интенсивность "
        "обслуживания μ. Возвращает коэффициент загрузки, среднее число "
        "заказов в системе, среднее время ожидания и среднее время пребывания "
        "заказа в системе. История расчетов на сервере не сохраняется."
    ),
)
def analyze_queue_system(request: QueueAnalysisRequest) -> QueueAnalysisResponse:
    return analyze_queue(request)


@router.get(
    "/formulas",
    response_model=QueueFormulasResponse,
    summary="Получить формулы модели M/M/1",
    description=(
        "Возвращает описание математической модели и формулы, используемые "
        "при анализе одноканальной системы массового обслуживания."
    ),
)
def get_queue_formulas() -> QueueFormulasResponse:
    return QueueFormulasResponse(
        model_name="M/M/1",
        description=(
            "Одноканальная система массового обслуживания с пуассоновским "
            "потоком заявок и экспоненциальным временем обслуживания."
        ),
        stability_condition="λ < μ",
        formulas={
            "rho": "ρ = λ / μ",
            "average_orders_in_system": "L = λ / (μ - λ)",
            "average_waiting_time": "Wq = λ / (μ * (μ - λ))",
            "average_time_in_system": "W = 1 / (μ - λ)",
        },
    )
